analyzer/os/photon/photon.go

```diff
@@ -3,53 +3,57 @@ package photon
 import (
 	"bufio"
 	"bytes"
+	"os"
 	"strings"
 
+	"github.com/aquasecurity/fanal/utils"
+
 	"github.com/aquasecurity/fanal/types"
 
 	"golang.org/x/xerrors"
 
-	"github.com/aquasecurity/fanal/analyzer/os"
+	aos "github.com/aquasecurity/fanal/analyzer/os"
 
 	"github.com/aquasecurity/fanal/analyzer"
-	"github.com/aquasecurity/fanal/extractor"
 )
 
 func init() {
-	analyzer.RegisterOSAnalyzer(&photonOSAnalyzer{})
+	analyzer.RegisterAnalyzer(&photonOSAnalyzer{})
+}
+
+var requiredFiles = []string{
+	"usr/lib/os-release",
+	"etc/os-release",
 }
 
 type photonOSAnalyzer struct{}
 
-func (a photonOSAnalyzer) Analyze(fileMap extractor.FileMap) (types.OS, error) {
-	for _, filename := range a.RequiredFiles() {
-		file, ok := fileMap[filename]
-		if !ok {
+func (a photonOSAnalyzer) Analyze(content []byte) (analyzer.AnalyzeReturn, error) {
+	photonName := ""
+	scanner := bufio.NewScanner(bytes.NewBuffer(content))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "NAME=\"VMware Photon") {
+			photonName = aos.Photon
 			continue
 		}
-		photonName := ""
-		scanner := bufio.NewScanner(bytes.NewBuffer(file))
-		for scanner.Scan() {
-			line := scanner.Text()
-			if strings.HasPrefix(line, "NAME=\"VMware Photon") {
-				photonName = os.Photon
-				continue
-			}
-
-			if photonName != "" && strings.HasPrefix(line, "VERSION_ID=") {
-				return types.OS{
+
+		if photonName != "" && strings.HasPrefix(line, "VERSION_ID=") {
+			return analyzer.AnalyzeReturn{
+				OS: types.OS{
 					Family: photonName,
 					Name:   strings.TrimSpace(line[11:]),
```

Comment:
What happens if `line` doesn't have enough length to do [11:]?

Reply:
Just empty.
https://play.golang.org/p/UclQNFPuCR_O